contracts/p1/Distributor.sol

```diff
@@ -94,24 +114,23 @@ contract DistributorP1 is ComponentP1, IDistributor {
         require(caller == rsrTrader || caller == rTokenTrader, "RevenueTraders only");
         require(erc20 == rsr || erc20 == rToken, "RSR or RToken");
         bool isRSR = erc20 == rsr; // if false: isRToken
+
         uint256 tokensPerShare;
+        uint256 totalShares;
         {
             RevenueTotals memory revTotals = totals();
-            uint256 totalShares = isRSR ? revTotals.rsrTotal : revTotals.rTokenTotal;
+            totalShares = isRSR ? revTotals.rsrTotal : revTotals.rTokenTotal;
             if (totalShares != 0) tokensPerShare = amount / totalShares;
             require(tokensPerShare != 0, "nothing to distribute");
         }
-
         // Evenly distribute revenue tokens per distribution share.
         // This rounds "early", and that's deliberate!
 
         Transfer[] memory transfers = new Transfer[](destinations.length());
```

Comment:
this needs to be `destinations.length() + 1`

Reply:
but actually I think it might be simpler to keep it size `destinations.length()` and just move the DAO fee block after the transfers. don't actually need to save it since we know it's the last one